qemu/qom_hierarchy.py

```diff
@@ -94,6 +91,35 @@ def find(self, **request):
         for t in co_find_eq(self.root().descendants(), **request):
             yield t
 
+    def merge(self, tree):
+        # cache
+        children = self.children
+        macros = self.macros
+        add_child = self.add_child
+
+        for m in tree.macros:
+            if m not in macros:
+                macros.append(m)
+
+        self.arches.update(tree.arches)
+
+        for n, other_c in tree.children.items():
+
+            if n in children:
+                c = children[n]
+            else:
+                c = type(other_c)(
+                    name = n,
+                    macros = other_c.macros,
+                    arches = other_c.arches,
+                )
+                add_child(c)
```

Comment:
Is this much faster than specifying a parent?
```
                c = type(other_c)(
                    name = n,
                    parent = self,
                    macros = other_c.macros,
                    arches = other_c.arches,
                )

Reply:
Don't know. U may try it on devel.

Reply:
(...same...)